packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -4191,6 +4209,22 @@ class SemanticsConfiguration {
   set isChecked(bool? value) {
     _setFlag(SemanticsFlag.hasCheckedState, true);
     _setFlag(SemanticsFlag.isChecked, value!);
+    assert(value != true || isCheckStateMixed != true);
```

Comment:
Make the assert a precondition check at the top of the setter to prevent it from getting into a bad state if the user violates an invariant.